src/main.h

```diff
@@ -134,6 +134,8 @@ extern CCriticalSection cs_main;
 extern CTxMemPool mempool;
 typedef boost::unordered_map<uint256, CBlockIndex*, ObjectHasher> BlockMap;
 extern BlockMap mapBlockIndex;
+typedef boost::unordered_map<uint256, int, ObjectHasher> ScCumTreeRootMap;
+extern ScCumTreeRootMap mapCumtreeHeight;
```

Comment:
That's a good point. This is indeed an ever growing map. Right now we just store info on blocks whose height is compatible with non ceasing sidechains, but this might be optimized further as you are suggesting (even though the map itself should be quite small in memory in absolute an relative terms, compared to the overall memory usage of zend).

I'd suggest to leave this as it is for now, an elaborate on optimization options in future.